### PR TITLE
Deploy app to gh-pages on successful build

### LINKED ISSUE #63: Deploy new react SUSI chat app on GitHub pages
- Deploy new chat app on Github pages.
- Update the README file

### PR DESCRIPTION
**Pull request by AnshulMalik**
Resolves #63 

Need to add deploy key to this repo and environment variable to travis.

### CODE REVIEW

File: deploy.sh [hunk 1]
```diff
@@ -47,12 +46,15 @@ git checkout $SOURCE_BRANCH
...
 # Actual building and setup of current push or PR.
 npm install
-npm run predeploy
+npm run build
+mv build ../build/
+
 git checkout $TARGET_BRANCH
-mv build/* .
+rm -rf node_modules/
+mv ../build/* .
```
Comment (rishiraj824) [inline deploy.sh]: `./build` is in the root directory what is the need for `../build`

File: deploy.sh [hunk 2]
```diff
@@ -14,16 +14,15 @@ REPO=`git config remote.origin.url`
...
-# Deceyption of the deploy_key.enc
 ENCRYPTED_KEY_VAR="encrypted_${ENCRYPTION_LABEL}_key"
 ENCRYPTED_IV_VAR="encrypted_${ENCRYPTION_LABEL}_iv"
 ENCRYPTED_KEY=${!ENCRYPTED_KEY_VAR}
 ENCRYPTED_IV=${!ENCRYPTED_IV_VAR}
-openssl aes-256-cbc -K $ENCRYPTED_KEY -iv $ENCRYPTED_IV -in deploy_key.enc -out deploy_key -d
+openssl aes-256-cbc -K $ENCRYPTED_KEY -iv $ENCRYPTED_IV -in deploy_key.enc -out ../deploy_key -d
 
-chmod 600 deploy_key
+chmod 600 ../deploy_key
```
Comment (rishiraj824) [inline deploy.sh]: `../deploy_key` is also in the root directory.

File: package.json [hunk 3]
```diff
@@ -29,7 +29,6 @@
     "build": "react-scripts build",
     "test": "react-scripts test --env=jsdom",
     "eject": "react-scripts eject",
-    "predeploy": "npm run build",
```
Comment (rishiraj824) [inline package.json]: Please do not delete this line, it's required to run the build process.

File: deploy.sh (hunk 1 above)
Comment (AnshulMalik) [inline deploy.sh]: When we do git add, this build folder gets added as well, so I moved it one directory up and then copied it's files back.

File: deploy.sh (hunk 2 above)
Comment (AnshulMalik) [inline deploy.sh]: Again, we don't want this key to be added to git, so moved one directory up.

File: package.json (hunk 3 above)
Comment (AnshulMalik) [inline package.json]: But I changed the build process.

File: deploy.sh (hunk 2 above)
Comment (rishiraj824) [inline deploy.sh]: @AnshulMalik Please do not add thet deploy_key here. It is supposed to be added in the repository, in *Settings*

File: package.json (hunk 3 above)
Comment (rishiraj824) [inline package.json]: @AnshulMalik Let this be there, for normal build processes which one could use during testing.

File: deploy.sh (hunk 2 above)
Comment (AnshulMalik) [inline deploy.sh]: There are two parts of a key, Private and Public, the public key gets added to repository settings, and private is kept here encrypted.

Comment (AnshulMalik) [inline deploy.sh]: It is decrypted by travis after build process completes and used to push to this repository's gh-pages branch. Hope it makes sense.

File: package.json (hunk 3 above)
Comment (AnshulMalik) [inline package.json]: I think is used for purge deploy, so I put it back there. Thanks

**Comment by AnshulMalik** [comment]
Hey @mariobehling, mind taking a look at it ?

**Pull request merged**